rest-server/src/dk/localghost/hold17/rest/api/hangman/HangmanRest.java

```diff
@@ -0,0 +1,103 @@
+package dk.localghost.hold17.rest.api.hangman;
+
+import com.google.gson.Gson;
+import dk.localghost.authwrapper.transport.AuthenticationException;
+import dk.localghost.hold17.dto.HangmanGame;
+import dk.localghost.hold17.dto.Token;
+import dk.localghost.hold17.helpers.HangmanHelper;
+import dk.localghost.hold17.helpers.TokenHelper;
+import dk.localghost.hold17.rest.auth.AuthenticationEndpoint;
+import dk.localghost.hold17.transport.IHangman;
+
+import javax.lang.model.type.ErrorType;
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.*;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+@Path("hangman")
+public class HangmanRest {
+        private static final Gson gson = new Gson();
+
+    //Start new game
+    @AuthenticationEndpoint.Auth
+    @POST
+    @Path("game")
+    @Produces(MediaType.TEXT_XML)
+    public Response startGame(@Context HttpServletRequest servletRequest) {
+        final String header = servletRequest.getHeader(HttpHeaders.AUTHORIZATION);
+        final String tokenStr = header.substring("Bearer ".length()).trim();
+
+        Token token = new Token();
+        token.setAccess_token(tokenStr);
+
+        IHangman hangman = HangmanHelper.getHangmanService(token, false);
+        try {
+            hangman.reset(token);
+        } catch(AuthenticationException ex) {
+
+        }
+        HangmanGame game = new HangmanGame();
+        game.setGame(hangman);
+        return Response
+                .ok("New game created by user ")
+                .build();
+    }
+
+    //Get game
+    @AuthenticationEndpoint.Auth
+    @GET
+    @Path("game")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getGame(@Context HttpServletRequest servletRequest) {
+        final String header = servletRequest.getHeader(HttpHeaders.AUTHORIZATION);
+        final String tokenStr = header.substring("Bearer ".length()).trim();
+
+        Token token = new Token();
+        token.setAccess_token(tokenStr);
+
+        IHangman hangman = HangmanHelper.getHangmanService(token, true);
+        if(hangman == null) {
+            String errMsg = "A game has not yet been created.";
+            return Response.status(400).entity(errMsg).build();
```

Comment:
Brug ErrorObj i stedet for en String. Derved får man stadig et JSON-objekt at arbejde med